src/click/decorators.py

```diff
@@ -339,22 +339,14 @@ def version_option(
         message = _("%(prog)s, version %(version)s")
 
     if version is None and package_name is None:
-        frame = inspect.currentframe()
-        assert frame is not None
-        assert frame.f_back is not None
-        f_globals = frame.f_back.f_globals if frame is not None else None
-        # break reference cycle
-        # https://docs.python.org/3/library/inspect.html#the-interpreter-stack
-        del frame
-
-        if f_globals is not None:
-            package_name = f_globals.get("__name__")
-
-            if package_name == "__main__":
-                package_name = f_globals.get("__package__")
-
-            if package_name:
-                package_name = package_name.partition(".")[0]
+        f_globals = inspect.currentframe().f_back.f_globals
```

Comment:
```tox -e typing``` seems to complain unless I check every step of this assignment

Reply:
It seems to break it up might be the solution
```python3
        frame = inspect.currentframe()

        if frame is not None:
            f_b = frame.f_back

        if f_b is not None:
            f_globals = f_b.f_globals

        del frame
```

Reply:
or perhaps

```python3
        frame = inspect.currentframe()
        assert frame is not None
        assert frame.f_back is not None
        assert frame.f_back.f_globals is not None
        f_globals = frame.f_back.f_globals
        # break reference cycle
        # https://docs.python.org/3/library/inspect.html#the-interpreter-stack
        del frame
```

Coverity's issue was that there were null checks after asserting the value was not **```None```**

Reply:
Which of these would be more acceptable?

Reply:
I think this might be better
```python3
        frame = inspect.currentframe()
        f_back = frame.f_back if frame is not None else None
        f_globals = f_back.f_globals if f_back is not None else None
        # break reference cycle
        # https://docs.python.org/3/library/inspect.html#the-interpreter-stack
        del frame
```

Reply:
I agree, that looks good.